File: src/types/skill.ts
export interface Skill {
  id: string;
  name: string;
  description?: string;
  path: string;
  version?: string;
  category: SkillCategory;
  icon?: string;
  isBuiltin: boolean;
  isEnabled: boolean;
  config: Record<string, unknown>;
  readmeContent?: string;
  installedAt: string;
}

export type SkillCategory =
  | 'document'
  | 'code'
  | 'data'
  | 'design'
  | 'ai'
  | 'integration'
  | 'custom';

export const SKILL_CATEGORY_COLORS: Record<SkillCategory, string> = {
  document: 'var(--accent-cyan)',
  code: 'var(--accent-primary)',
  data: 'var(--accent-success)',
  design: 'var(--accent-purple)',
  ai: 'var(--accent-orange)',
  integration: 'var(--accent-warning)',
  custom: 'var(--text-secondary)',
};

File: src-tauri/src/models/skill.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Skill {
    pub id: String,
    pub name: String,
    pub description: Option<String>,
    pub path: String,
    pub version: Option<String>,
    pub category: Option<String>,
    pub icon: Option<String>,
    pub is_builtin: bool,
    pub is_enabled: bool,
    pub config_json: String,
    pub readme_content: Option<String>,
    pub installed_at: String,
}
